util/http_client_pool.go

```diff
@@ -37,7 +41,30 @@ func init() {
 }
 
 func GetHttpClient() *http.Client { return httpClientPool.Get().(*http.Client) }
+func GetHttpClientForOauth2(oauthToken string) (*http.Client, error) {
+	if oauthToken != "" {
+		splitToken := strings.Split(oauthToken, " ")
+		if len(splitToken) != 2 {
+			return nil, errors.New("token format was invalid, expected 'token [token]' or empty token")
+		}
+		oauthToken = splitToken[1]
+	}
+	client := httpClientPool.Get().(*http.Client)
+	client.Transport = &oauth2.Transport{
+		Base: client.Transport,
+		Source: oauth2.ReuseTokenSource(nil, oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: oauthToken},
+		)),
+	}
+
+	return client, nil
+
+}
+
 func PutHttpClient(c *http.Client) {
+	if oauthTransport, ok := c.Transport.(*oauth2.Transport); ok {
```

Comment:
I'd write an oauth-specific put? just because this could get confusing. 

other options would be to: 
- not put these clients back into the pool
- have a seperate oauth pool.

Reply:
done, went with oauth specific put.

Reply:
There's a separate get, but not a separate put.